Extend Log RBAC with Logs Pipelines, Indexes and Archives best practices. (#8698)
DataDog/documentation: content/en/logs/guide/logs-rbac.md

``` bash
curl -X DELETE "https://api.datadoghq.com/api/v2/roles/<ROLE_ID>/users" -H "Content-Type: application/json" -H "DD-API-KEY: <DATADOG_API_KEY>" -H "DD-APPLICATION-KEY: <DATADOG_APP_KEY>" -d '{"data": {"type":"users","id":"<USER_ID>"}}'
```

[1]: /api/v2/users/#list-all-users
[2]: /api/v2/roles/#add-a-user-to-a-role
[3]: /api/v2/roles/#remove-a-user-from-a-role

{{% /tab %}}
{{< /tabs >}}

## Restrict access to logs

This section explains how to grant ACME Team members (both `ACME Admin` and `ACME User` members) access to `team:acme` logs—and only `team:acme` logs. It uses the [Log Read Data][74] permission scoped with Restriction Queries.

As a good practice for maximum granularity and easier maintenance, you should **not** extend the permissions of ACME users to access more logs. Do not restrict other roles to the same `team:acme` restriction query. Instead, consider assigning users to multiple roles based on what each of them, individually, needs to access.

This section details how to:

1. Create a `team:acme` restriction query.
2. Attach that restriction query to ACME roles.

> *Note*: Roles can have **no more than one** restriction query attached. If you attach a restriction query to a role, it also removes whichever restriction query already attached to this role.

Reviewer: ```suggestion
*Note*: Roles can have **no more than one** restriction query attached. If you attach a restriction query to a role, it removes any restriction queries already attached to this role.
```

{{< img src="logs/guide/rbac/indexes.png" alt="ACME Indexes"  style="width:60%;">}}

### Log Archives

#### Read Log Archives

Create one or multiple [archives][62] for `team:acme` logs. Assign the [Read Archives][72] permission to members of `ACME Admin`, but **scoped** to that ACME Archive(s).

{{< img src="logs/guide/rbac/archives.png" alt="ACME Archives"  style="width:60%;">}}

Multiple archives can be useful if you have different lifecycle policies depending on logs (for instance, for production and staging logs). Keep in mind that rehydration is intended to work for only one archive at a time, though you can trigger multiple rehydrations on multiple archives at once. 

#### Write Historical Views

Assign the [Write Historical View][73] permission to members of `ACME Admin`. This permission grants the ability to perform rehydrations. 

> **Optionally**, set up your Log Archives so that all logs rehydrated from that archive will eventually have the `team:acme` tag, whether whether or not they had the tag in the archive. This option enables you to enforce consistency with your existing restriction policies, as well as to safely remove deprecated restrictions that correspond to no more logs flowing in Datadog or indexed in Datadog.

Reviewer: ```suggestion
**Optionally**, set up your Log Archives so that all logs rehydrated from that archive will eventually have the `team:acme` tag, whether or not they had the tag in the archive. This option enables you to enforce consistency with your existing restriction policies, as well as to safely remove deprecated restrictions that correspond to no more logs flowing in Datadog or indexed in Datadog.
```
